src/group_theory/perm/sign.lean

```diff
@@ -86,6 +86,28 @@ f.subtype_perm (λ x, ⟨h x, λ h₂, f.inv_apply_self x ▸ perm_inv_on_of_per
   (h : ∀ x, p x → p ((1 : perm α) x)) : @subtype_perm_of_fintype α 1 p _ h = 1 :=
 equiv.ext $ λ ⟨_, _⟩, rfl
 
+lemma perm_on_inr_of_perm_on_inl {m n : Type u} [fintype m] [fintype n] (σ : equiv.perm (m ⊕ n))
+  (h : ∀ a1, ∃ a2, sum.inl a2 = σ (sum.inl a1)) : ∀ b1, ∃ b2, sum.inr b2 = σ (sum.inr b1) :=
```

Comment:
I think this lemma amounts to saying that `∃ σl σr, σ = perm.sum_congr σl σr`. I wonder if you can avoid needing it by working with `perm.sum_congr` directly...

Reply:
Sorry, but I don't see how. Not every `σ = perm.sum_congr σl σr` for some `σl` and `σr`, so the question is what is the least you can assume to deduce that is can be so written. This lemma gives one answer. Also note that the lemma isn't true if `m` and `n` aren't `fintypes`.